New build warning

I presume this is from 3b6fd5d494
```
keyring2john.c: In function 'process_file':
keyring2john.c:161:12: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  file_name = fname;
            ^
```
@loverszhaokai please watch out for things like this. Always build using `make -s` so you see the warnings (and not just a wall of text).

--- a/src/keyring2john.c
+++ b/src/keyring2john.c
@@ -27,7 +27,7 @@ typedef unsigned char guchar;
 typedef unsigned int guint;
 typedef int gint;
 static int count;
-char *file_name;
+const char *file_name;
 
 static void warn_exit(const char *fmt, ...)
 {

